xcode/build_phases/swiftlint.rb

```diff
@@ -0,0 +1,393 @@
+#https://github.com/TouchInstinct/Styleguide/blob/multiple_swiftlint/IOS/Guides/BuildScripts/Multiple_Swiftlint_Guide.md
+require 'yaml'
+require 'optparse'
+require 'ostruct'
+require 'date'
+require 'fileutils'
+
+Encoding.default_external = Encoding::UTF_8
+Encoding.default_internal = Encoding::UTF_8
+
+class String
+    def with_wrapped_whitespace
+        self.gsub(/\s+/, '\ ')
+    end
+    
+    def filter_allowed_symbol_into_path
+        self.gsub!(/[^0-9A-Za-z \-+.\/]/, '')
+    end
+  
+    def true?
+        self.to_s.downcase == "true"
+    end
+    
+    def add_back_to_path(count)
+        string = self
+        count.to_i.times { |i|
+            string = '../' + string
+        }
+        return string
+    end
+    
+    def nilOrEmpty?
+        self.nil? or self.empty?
+    end
+end
+
+class Array
+    def nilOrEmpty?
+        self.nil? or self.empty?
+    end
+end
+
+class YamlManager
+    def initialize(swiftlint_yaml_path)
+        @swiftlint_yaml_path = swiftlint_yaml_path
+        @configuration ||= YAML.load(File.read(@swiftlint_yaml_path))
+    end
+    
+    def get_configuration(key)
+        @configuration[key]
+    end
+
+    def update(key, new_configuration_values)
+        @configuration[key] = new_configuration_values
+        save_settings(@configuration)
+    end
+
+    private
+
+    def save_settings(settings)
+        File.write(@swiftlint_yaml_path, settings.to_yaml)
+    end
+end
+
+class SettingOption
+    def initialize
+        @options = OpenStruct.new
+        OptionParser.new do |opt|
+            opt.on('-s', '--source_directory STRING', 'The directory of source') { |option| @options.source_directory = option }
+            opt.on('-p', '--pods_directory STRING', 'The directory of pods') { |option| @options.pods_directory = option }
```

Comment:
--exluded_dirs (Pods, Generated, etc.)

Reply:
got it. you need --swiftlint_executable_path

Reply:
>or you suggest to change naming so pods_directory -> swiftlint_executable_path

yep. it's not clear why I need to pass pods_directory. my first guess was to exclude it from linting

Reply:
`swiftlint_executable_path` is a bad naming because it doesn't contain the full path to `SwiftLint`